src/components/common/ErrorAlert.js

```diff
@@ -1,19 +1,31 @@
-import React from 'react';
-import { Alert } from 'react-bootstrap';
+import React, { useEffect, useRef } from 'react';
 import useUser from '../../contexts/UserContext';
+import useNotes from '../../contexts/NotesContext';
+import { useAlert } from 'react-alert';
 
 const ErrorAlert = () => {
-  const [user, dispatch] = useUser();
+  const alert = useAlert();
+  const [user, dispatchUser] = useUser();
+  const [notes, dispatchNotes] = useNotes();
 
-  if (user.error) {
-    setTimeout(() => dispatch({ type: 'CLEAR_ERROR' }), 5000);
-    return (
-      <Alert variant='primary' className='mt-3 mb-0'>
-        <strong>We have encountered an error!</strong> {user.error.message}
-      </Alert>
-    );
-  }
-  return null;
+  const isFirstRun = useRef(true);
+
+  useEffect(() => {
+    if (isFirstRun.current) isFirstRun.current = false;
+    else {
+      if (user.error) {
+        setTimeout(() => dispatchUser({ type: 'CLEAR_ERROR' }), 3000);
+        alert.error(user.error.message);
+      }
+
+      if (notes.error) {
+        setTimeout(() => dispatchNotes({ type: 'CLEAR_ERROR' }), 3000);
+        alert.error(notes.error.message);
+      }
+    }
+  }, [user, notes, alert, dispatchUser, dispatchNotes]);
```

Comment:
If you want to show error only when user.error or note.error changes, then wouldn't it be better to have only those dependencies in useEffect array? To see how to hack eslint useEffect error look at other useEffects.